main.tf

```diff
@@ -20,3 +23,15 @@ module "records" {
   ]
   allow_overwrite = var.allow_overwrite
 }
+
+module "records_data" {
+  source   = "./modules/record_collection_data"
+  for_each = local.collections_data
+
+  zone_id = data.cloudflare_zone.this.id
+  name    = each.value[0].name
+  type    = each.value[0].type
+  data    = each.value[0].data
```

Comment:
You have to iterate here. In case you have multiple records with different data blocks.

Reply:
I already tried this and only one data block is allowed.

Reply:
No, I mean not multiple blocks in one resource, but multimple records of type `CAA` with the same host name.
For example:

```
    {
      name     = "somedomain"
      type     = "CAA"
      ttl      = 120
      data     = {
        flags = 0
        tag   = "issue"
        value = "someca.com"
      }
    },
    {
      name     = "somedomain"
      type     = "CAA"
      ttl      = 120
      data     = {
        flags = 1
        tag   = "issue"
        value = "anotherca.com"
      }
    },
```

Reply:
I see. Good catch. Fixed!